streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -580,4 +562,118 @@ private String getBrokerSideConfigValue(final Config brokerSideTopicConfig,
 
         return topicsToCreate;
     }
+
+    /**
+     * Sets up internal topics.
+     *
+     * Either the given topic are all created or the method fails with an exception.
+     *
+     * @param topicConfigs internal topics to setup
+     */
+    public void setup(final Map<String, InternalTopicConfig> topicConfigs) {
```

Comment:
should we make an effort to clean up created topics on failure? currently this method is not retriable